### PR TITLE
Feat/floating UI

### LINKED ISSUE #2783: ActionGroup creates scrollable overflow and hides below table
### Package

filament/filament

### Package Version

v2.13.2

### Laravel Version

v9.17

### Livewire Version

v2.10.5

### PHP Version

PHP 8.1.7

### Bug description

Expanded ActionGroup is restricted to table container's bounds, so any potential overflow creates scrollbar and actions are pushed under it. Picture below:

1:
Zrzut ekranu z 2022-06-15 22-38-33

2:
Zrzut ekranu z 2022-06-15 22-30-52

### Steps to reproduce

- Create action group and add actions to it
- Propagate enough data for table to render
- Open ActionGroup in table's last row/record

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by awcodes**
Add Floating UI to handle drop downs and popups. Allows for customizable placement of panels as well as auto flipping, etc to avoid overflow hidden issues. Also traps focus to opened items using x-trap.

Uses [@awcodes/alpine-floating-ui](https://www.npmjs.com/package/@awcodes/alpine-floating-ui) under the hood.

This was previously discussed with @danharrin.

### CODE REVIEW

**Comment by themahdavi** [comment]
image

Hello @awcodes 
by default first item focus when open dropdown, I think not bug at all
and Tab functionality is work fine 👌

**Comment by kamilst96** [comment]
@awcodes it looks as if you opened PR from #2825, instead of 2.x. That PR has already been merged, so if you fetch upstream 2.x on your fork and then rebase PR branch against it, it should be gone.

**Comment by danharrin** [comment]
I've fixed it

**Comment by awcodes** [comment]
Thanks @danharrin. I’m trying to keep this in sync for you with the new releases so it’s ready to go when you want it.

**Comment by danharrin** [comment]
I'm reviewing it now so it should be fine :)

File: packages/forms/resources/views/components/date-time-picker.blade.php [hunk 1]
```diff
@@ -35,9 +36,9 @@
         <button
             @unless($isDisabled())
                 x-ref="button"
-                x-on:click="$float({placement: 'bottom-start', offset: 8, flip: {} }, {trap: true}); openPicker()"
-                x-on:keydown.enter.stop.prevent="open ? selectDate() : openPicker()"
-                x-on:keydown.arrow-left.stop.prevent="focusPreviousDay()"
+                x-on:click="togglePanelVisibility()"
+                x-on:keydown.enter.stop.prevent="isOpen() ? selectDate() : togglePanelVisibility()"
+                x-on:keydown.arrow-left.stop.prevent="console.log('hey'); focusPreviousDay()"
```
Comment (saade) [inline packages/forms/resources/views/components/date-time-picker.blade.php:41]: hey!

**Review by saade — COMMENTED** [review]
Better implementation than mine! Great work @awcodes!

File: packages/forms/resources/views/components/date-time-picker.blade.php (hunk 1 above)
Comment (danharrin) [reply packages/forms/resources/views/components/date-time-picker.blade.php:41]: whoops 😂

**Comment by danharrin** [comment]
@awcodes When you finish refactoring to the directive, mark this as ready for review. No rush 😅

**Comment by danharrin** [comment]
Thank you very much!!

**Comment by HomaEEE** [comment]
after update to new version v2.13.28 table with one row: 
Снимок экрана 2022-07-17 в 17 45 16

and two row: 
Снимок экрана 2022-07-17 в 17 47 11
Снимок экрана 2022-07-17 в 17 47 03

**Comment by saade** [comment]
> after update to new version v2.13.28 table with one row: Снимок экрана 2022-07-17 в 17 45 16
> 
> and two row: Снимок экрана 2022-07-17 в 17 47 11 Снимок экрана 2022-07-17 в 17 47 03

thats expected... since it does not teleport the dropdown outside the overflowed container ... @danharrin ?

**Comment by danharrin** [comment]
Indeed, it should be `overflow-x-auto` as this is more appropriate, but unfortunately the behaviour is the same. That will not fix the issue as browsers will still hide the y overflow.

**Comment by awcodes** [comment]
You're both right. I do have an idea to try, if you can give me a day to work it out.

**Comment by danharrin** [comment]
Of course! Take your time mate :)

**Comment by awcodes** [comment]
PR has been submitted with a fix for this. 

Thank you both.

**Pull request merged**